### PR TITLE
Add PowerLawPrior, fix other priors and add test for priors

### PR DESCRIPTION
**Pull request by xuyuon**

### CODE REVIEW

File: src/jimgw/prior.py [hunk 1]
```diff
@@ -270,10 +272,11 @@ def __init__(self, parameter_names: list[str]):
         self.parameter_names = parameter_names
         assert self.n_dim == 1, "SinePrior needs to be 1D distributions"
         super().__init__(
-            UniformPrior(-1.0, 1.0, [f"cos_{self.parameter_names[0]}"]),
+            CosinePrior([f"{self.parameter_names[0]}"]),
```
Comment (kazewong) [inline src/jimgw/prior.py:275]: Why is this Cosine here

Comment (kazewong) [reply src/jimgw/prior.py:275]: NVM I see the offset by pi /2. Can you add a line on this is for bypassing gradient errors?

File: test/test_prior.py [hunk 2]
```diff
@@ -1,23 +1,115 @@
...
+        assert jnp.all(jnp.isfinite(samples['x_phi']))
+        # Check that the log_prob is finite
+        samples = {}
+        for i in range(3):
+            samples.update(trace_prior_parent(p, [])[i].sample(jax.random.PRNGKey(0), 10000))
+        log_prob = jax.vmap(p.log_prob)(samples)
+        assert jnp.all(jnp.isfinite(log_prob))
+    
+    def test_power_law(self):
+        from bilby.core.prior.analytical import PowerLaw
```
Comment (kazewong) [inline test/test_prior.py:83]: Bilby is not a dependence of jim, and I don't think we should add it even in the test. Please remove this

**Review by kazewong — COMMENTED** [review]
Please address the comments in this review

**Review by kazewong — APPROVED** [review]
Looks good now
